### ISSUE TITLE
Fine tuning for opinion polarity?

### ISSUE THREAD

**Issue by youssefavx** [UKPLab/sentence-transformers#703]
Thanks so much for your helpful training guides! 

I want to fine tune a model to be able to bring opposing views/values to the top like so:

1. I’m a democrat
2. I’m a republican 
Score: -1

1. I’m a democrat 
2. I hate politics 
Score: -1

1. I hate marshmallows 
2. I love marshmallows 
Score: -1

Now I’m not sure if 
1. I can use -1 (or I should go with 1)
2. if the loss function is going to be cosine similarity if that’s going to be appropriate for a task like this during training 
3. What the best way to go about this is / which models to consider 

Would love any help whatsoever and am very appreciative of all the work you put into this to make it such a breeze!

**Comment by youssefavx** [comment]
Actually looking through the documentation it seems something like Contrastive loss might be better?

**Comment by nreimers** [comment]
Yes, you can use -1 as label and use either CosineSimilarityLoss or Constrative Loss.

Note: Achieving -1 for opposing views can be quite hard, as "I have marshmallows" and "I love marshmallows" are semantically fairly similar and a -1 indicates that these are in completely opposite directions in the vector space.

**Comment by liebkne** [comment]
@nreimers How is CosineSimilarityLoss different from Constrative Loss in binary classification?

**Comment by youssefavx** [comment]
@nreimers Thank you! I'm hoping to do it so it's continuous / not classification. So I wouldn't want it to classify it by -1, but to have degrees (the more opposing the views, the closer it is to -1 (or some other value)), I assume this is what happens when I measure the cosine similarity on the embeddings of 2 separate sentences after training with contrastive loss?

**Comment by youssefavx** [comment]
Also I assume one way of fixing the "I have marshmallows" and "I love marshmallows" thing would be to set the score to 0 to increase the distance of the embeddings of those kinds of sentences? And would that apply to sentences that are unrelated to the same topic? Such as "I ate peanut butter last night" and "I went to the movies" (setting irrelevant sentences to 0)?
